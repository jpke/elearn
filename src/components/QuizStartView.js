import React from 'react';
import highScore from '../utils/highScore';

const QuizStartView = (props) => (
      <div className="quiz-start">
        {props.attempts.length ? (
          <div>
            <h3>{props.title}</h3>
            <h3>You have attempted this quiz {
                props.attempts.length === 1 ?
                  props.attempts.length.toString().concat(" time")
                  :
                  props.attempts.length.toString().concat(" times")
                }
            </h3>
            <h4>Your best score was {highScore(props.attempts)}</h4>
            <h3>You have {props.passed ? "" : "not"} passed this quiz</h3>
          </div>
        ):(
          <h3>You have not yet attempted this quiz</h3>
        )
      }
      <div className="quiz-start-button-container">
        <button className="quiz-start-button" onClick={props.startQuiz}> Start Quiz </button>
        <button className="quiz-start-button" onClick={props.viewQuizzes}>Back to Quiz List</button>
      </div>
    </div>
);

export default QuizStartView;
