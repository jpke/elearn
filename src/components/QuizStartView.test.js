/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import QuizStartView from './QuizStartView';

let attempts;
beforeAll(() => {
  attempts = [
    {
      score: 0
    },
    {
      score: 2
    },
    {
      score: 1
    }
  ]
})

it('renders correctly for non passed quiz', () => {
  const tree = renderer.create(
    <QuizStartView
      startQuiz={jest.fn()}
      viewQuizzes={jest.fn()}
      passed={false}
      score={5}
      attempts={attempts}
      title={"sample quiz title"}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for passed quiz', () => {
  const tree = renderer.create(
    <QuizStartView
      startQuiz={jest.fn()}
      viewQuizzes={jest.fn()}
      passed={true}
      score={5}
      attempts={attempts}
      title={"sample quiz title"}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
