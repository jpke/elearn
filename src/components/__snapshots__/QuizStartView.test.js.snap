exports[`test renders correctly for non passed quiz 1`] = `
<div
  className="quiz-start">
  <div>
    <h2>
      sample quiz title
    </h2>
    <h4>
      Last score: 
      1
    </h4>
    <h4>
      Best score: 
      2
    </h4>
    <h4>
      Attempts: 
      3
    </h4>
    <h3>
      You have 
      not
       passed this quiz
    </h3>
  </div>
  <div
    className="quiz-start-button-container">
    <button
      className="quiz-start-button"
      onClick={[Function]}>
       Start Quiz 
    </button>
    <button
      className="quiz-start-button"
      onClick={[Function]}>
      Back to Quiz List
    </button>
  </div>
</div>
`;

exports[`test renders correctly for passed quiz 1`] = `
<div
  className="quiz-start">
  <div>
    <h2>
      sample quiz title
    </h2>
    <h4>
      Last score: 
      1
    </h4>
    <h4>
      Best score: 
      2
    </h4>
    <h4>
      Attempts: 
      3
    </h4>
    <h3>
      You have 
      
       passed this quiz
    </h3>
  </div>
  <div
    className="quiz-start-button-container">
    <button
      className="quiz-start-button"
      onClick={[Function]}>
       Start Quiz 
    </button>
    <button
      className="quiz-start-button"
      onClick={[Function]}>
      Back to Quiz List
    </button>
  </div>
</div>
`;
